### ISSUE TITLE
Polymorphic model relation doesn't work with `model: *`

### ISSUE THREAD

**Issue by jrybinski** [strapi/strapi#5348]
**Describe the bug**
I was trying to create a model with polymorphic relation. 
I have read docs - https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#relations

So I defined model and as I expected admin panel doesn't work. So I was trying to create one using API endpoint.
When I was trying to create it was throwing error. After I switch from polymorphic relation `*` into an exact define model then it was working fine.

**Steps to reproduce the behavior**
1. Go to Content Type Builder
2. Define new model with anykind of fields, for example `Favourite`
3. Go to created files and edit JSON schema file( `favourite.settings.json`)
4. In section, attributes add polymorphic field
```
{
  "connection": "default",
  "collectionName": "favourites",
  "info": {
    "name": "favourite"
  },
  "options": {
    "increments": true,
    "timestamps": true
  },
  "attributes": {
    "title": {
      "type": "string",
      "required": true
    },
    "contentItems": {
      "collection": "*",
      "filter": "field"
    }
  }
}
```
5. Try to post on endpoint `POST http://localhost:1337/favourites`
In array is id of `Event` object, I have `Events` collection and it's id tooked after created.

```
{
  "title":"Title1",
  "contentItems": ["5e56dc4d8d9153820f228bab"]
} 
```
6. You will get 
```
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "An internal server error occurred"
}
``` 
and in Strapi conosole
```
[2020-02-27T20:47:00.530Z] error TypeError: Cannot read property 'associations' of undefined
    at newValue.forEach.obj (/node_modules/strapi-connector-mongoose/lib/relations.js:234:51)
    at Array.forEach (<anonymous>)
    at params.parseRelationships.Object.keys.reduce (/node_modules/strapi-connector-mongoose/lib/relations.js:219:28)
    at Array.reduce (<anonymous>)
    at Function.update [as updateRelations] (/node_modules/strapi-connector-mongoose/lib/relations.js:38:59)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
7. Change polymorphic into exact defined collection
```
    "contentItems": {
      "collection": "*",
      "filter": "field"
    }
```
into 
```
    "contentItems": {
      "collection": "event",
      "filter": "field"
    }
```
8. Try to post same request
9. An object will be created with success.

**Expected behavior**
 An object will be created with success. I can put there any kind of model and it should work.

**System**

- Node.js version: 10.18.1
- NPM version: 6.13.4
- Strapi version: 3.0.0-beta.18.7
- Database: MongoDB
- Operating system: macOS Catalina

**Additional context**
I would love to help with fixing it.

**Comment by alexandrebodin** [comment]
Hi @jrybinski you need to send more infos here:

```json
{
  "title":"Title1",
  "contentItems": [{
    "ref": "target-model",
    "refId": "5e32de776ae1c39b89261c81"
  }]
} 
```

**Comment by jrybinski** [comment]
@alexandrebodin Thanks a lot! It works.
Is it documented anywhere?

**Comment by alexandrebodin** [comment]
Not right now. It is a bit buggy as of now and needs refactoring before we can document it :)

**Comment by nicholasbester** [comment]
Thanks @alexandrebodin when you say it's a bit buggy what would be the main concerns to watch out for?

**Comment by alexandrebodin** [comment]
If you are using an sql db you will need to have a reverse field that points to the morph relation.

**Comment by Istanful** [comment]
@alexandrebodin I have a similar problem. I intend to setup a `post` entity that has many `category` entities. A `category` must belong to either a `post` or a `category`, but never both. I setup an application with `--quick-start` and the server gives me an error message.

Am I going at this wrong or is there a bug?

**How to reproduce:**

In the `category.settings.json` I configure the category to have a `parent` with a polymorphic model. I also said that it has many `categories` via `parent`.

```json
{
    "parent": {
      "model": "*",
      "filter": "field"
    },
    "categories": {
      "collection": "category",
      "via": "parent"
    }
}
```

In the `post.settings.json` I configure the `post` to have many `category` via `parent`.

```json
{
    "categories": {
      "collection": "category",
      "via": "parent"
    }
}
```

When I start the development server I get the following error:

```text
[2020-03-11T12:44:34.983Z] error Something went wrong in the model `Category` with the attribute `parent`
[2020-03-11T12:44:34.984Z] error TypeError: Cannot read property 'nature' of undefined
```

- In `strapi-utils/lib/models.js:376` the code attemtps to fetch the `nature`.
- At `strapi-utils/lib/models.js:79` it loops through all models to find what model references the `category#parent`.
- At `strapi-utils/lib/models.js:84` I assume that it should find that a `post` has many `category` via `parent`. It does **not** because it expects the `via` key as well as the `model` key set to the current model.

According to the [documentation](https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#relations) however; only the collection key should be set.

If I change the `post.settings.json` to this the development server starts:

```json
{
    "categories": {
      "collection": "category",
      "model": "category",
      "via": "parent"
    }
}
```

It does however not allow me to set `parent` for category nor `categories` for `category`. I assume that the error is at `strapi-utils/lib/models.js:84` where the condition `attribute.model === currentModelName` should be removed.

**Comment by SuperMasterBlasterLaser** [comment]
I have also encountered this problem.

I have created separate plugin with content type that should be related to any other models.

When I add this

    "content_type": {
      "model": "*",
      "filter": "field"
    },

I get `TypeError: Cannot read property 'nature' of undefined`. I know that it will work if I use MongoDB. However, I wanted to make my plugin more universal.

Does that mean that I need to use Components instead of ContentTypes?

**Comment by SuperMasterBlasterLaser** [comment]
I have looked at source code of file upload https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-upload/models/File.settings.json#L72

How does it work, when I create plugin with content type that has same settings?

**Comment by SuperMasterBlasterLaser** [comment]
I see that this upload_file content type has created additional field `upload_file_morph` how to make strapi to generate this table implicitly?

**Comment by SuperMasterBlasterLaser** [comment]
@alexandrebodin 

Is that reversing explicitly required for morph tables? Then how file upload plugin creates this morph table?

**Comment by lauriejim** [comment]
It looks working with 
```json
"parent": {
  "collection": "*",
  "filter": "field"
}
```
Probably an issue w/ `model` instead of `collection`.

**Comment by derrickmehaffy** [comment]
It appears there is another issue trying to create entries in the Admin (even if you aren't setting any fields that are polymorphic)

[2021-01-22T19:46:55.469Z] error error: column "related_type" does not exist
    at Parser.parseErrorMessage (/home/dmehaffy/Development/test8607/node_modules/pg-protocol/dist/parser.js:278:15)
    at Parser.handlePacket (/home/dmehaffy/Development/test8607/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/home/dmehaffy/Development/test8607/node_modules/pg-protocol/dist/parser.js:39:38)
    at TLSSocket.<anonymous> (/home/dmehaffy/Development/test8607/node_modules/pg-protocol/dist/index.js:10:42)
    at TLSSocket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:298:12)
    at readableAddChunk (_stream_readable.js:273:9)
    at TLSSocket.Readable.push (_stream_readable.js:214:10)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:188:23)
From previous event:
    at Client_PG._query (/home/dmehaffy/Development/test8607/node_modules/knex/lib/dialects/postgres/index.js:234:12)
    at Client_PG.query (/home/dmehaffy/Development/test8607/node_modules/knex/lib/client.js:158:17)
    at /home/dmehaffy/Development/test8607/node_modules/knex/lib/transaction.js:331:24
From previous event:
    at Client_PG.trxClient.query (/home/dmehaffy/Development/test8607/node_modules/knex/lib/transaction.js:326:12)
    at Runner.query (/home/dmehaffy/Development/test8607/node_modules/knex/lib/runner.js:135:36)
    at /home/dmehaffy/Development/test8607/node_modules/knex/lib/runner.js:39:23
From previous event:
    at Runner.run (/home/dmehaffy/Development/test8607/node_modules/knex/lib/runner.js:25:16)
    at Builder.Target.then (/home/dmehaffy/Development/test8607/node_modules/knex/lib/interface.js:14:43)
    at processImmediate (internal/timers.js:461:21)

**Comment by ggirodda** [comment]
@derrickmehaffy same issue here. This is because I used another attribute name, instead of "related", I think it's a bug. If it's the same problem than mine, you have to rename the attribute to "related"
```javascript
   "related": {
      "collection": "*",
      "filter": "field"
    }
```
and on the other side
```javascript
   "someAttr": {
      "model": "pages",
      "via": "related"
   }
```

**Comment by Marc-Roig** [comment]
Hello :) 
We don't support this syntax anymore in V4 (V3 reached EOL), so in the effort to close issues that do not have a solution or are deprecated I will close this one too.
